documentation/src/docs/controller-service.md

```diff
@@ -122,6 +123,13 @@ Upon creating Transactions, Controller also tracks Transaction timeouts
 and aborts transactions whose timeouts have elapsed. Details of
 Transaction management can be found later in the [Transactions](#transaction-manager) section.
 
+  4. [**Watermarks**](http://pravega.io/docs/v0.8.0/watermarking/): Watermarks are a data structure produced by controller that refer to a unique position in the stream and associates time information corresponding to that position. The watermark is computed by controller by coordinating the time and position information reported by different writer applications and generating a time window. 
+
+## Key-Value-Tables Management
+
+Apart from stream abstraction controller is also the source of truth for the other storage primitive offered by pravega - Key Value Tables. Just like a stream, a key value table is also distributed and paritioned using a table segment. A table segment has same properties as a stream segment, but with the data is formatted as keys and values and indexed on the keys. It provides APIs to perform CRUD operations on keys. Controller uses table segments to create a higher level abstraction which creates a distributed key value table. Each table is partitioned and the user data is distributed across different partitions using a hashing scheme. It also introduces a concept of key family which is used to ensure that all keys from the same key family are always mapped to the same partition.   
+Controller maintains the metadata about the key value tables and is responsible for its lifecycle. Presently key value tables do not support any user defined policies like scaling or retention. So controller's role is limited to provisioning table segments for the table and managing its lifecycle which includes operations like create seal and delete tables.   
+
 ## Cluster Management
```

Comment:
We should add more detail here about how we handle node failovers and manage Controller and SSS nodes being added/removed from the cluster.

Reply:
This is overview document.. We should perhaps have separate documents detailing each aspect separately.

Reply:
Ok. It may be good file a documentation issue for this.